modules/openapi-generator/src/main/resources/C-libcurl/CMakeLists.txt.mustache

```diff
@@ -75,10 +81,15 @@ set(HDRS
 
 )
 
-# Add library with project file with projectname as library name
-add_library(${pkgName} SHARED ${SRCS} ${HDRS})
+include(PreTarget OPTIONAL)
+
+# Add library with project file with project name as library name
+add_library(${pkgName} ${SRCS} ${HDRS})
```

Comment:
Why need remove `SHARED` ?

Reply:
The SHARED/STATIC types should only be used if we know in advance that the library works when compiled as one and not as the other. By removing the flag I could add the -DBUILD_SHARED=FALSE flag to the CMake command and compile it as a static library which worked for me without any problem.
Is there any restriction for compiling as a static library?